service/internal/credentials/vault.go

```diff
@@ -191,37 +191,42 @@ func genProjectAppRole(name string) string {
 	return fmt.Sprintf("%s/%s-%s", vaultAppRolePrefix, vaultProjectPrefix, name)
 }
 
-func (v VaultProvider) CreateToken(name string) (string, string, string, error) {
+func (v VaultProvider) CreateToken(name string) (types.Token, error) {
```

Comment:
Opted to create a `types.Token` and return that versus returning a couple new strings for creation and expiration time. This also gives flexibility in the future in case other data is needed.